### ISSUE TITLE
Highlighting of Cilk keywords in Markdown code blocks

### ISSUE THREAD

**Issue by ailiop** [OpenCilk/www.opencilk.org#126]
### Summary

@OpenCilk/opencilk-developers I believe I have figured out how to enable highlighting of Cilk keywords in our code blocks.  The idea is to add Cilk support to [PrismJS](https://prismjs.com/), which I understand is the code highlighting engine used by [Eleventy](https://www.11ty.dev/docs/plugins/syntaxhighlight/).  The purpose of this issue is to touch base with the OpenCilk team before I submit a PR to the [Prism repo](https://github.com/PrismJS/prism).

### How would it work?

Our Markdown code blocks with Cilk code would be marked as, say, ` ```cilkc ` (instead of ` ```c `).

There is a separate question of "how do we make sure our site uses the updated PrismJS engine?".  I suspect it will be easy once the PR is merged to Prism — but mainly I hope @behoppe can take care of that.  :)

### Main questions

I seek feedback on the following questions as they are relevant to the *interface* that will be exposed to the user (e.g., ourselves).  Other details I believe are less important as they can be modified after the fact without disruption.

1. Should Cilk/C and Cilk/C++ be registered as separate languages?  And, if yes:
2. What should we name the Cilk language IDs?
    - Note: Prism allows *aliases* for language IDs

**Current approach.**  My working implementation adds two separate languages:
- `cilkc` (alias: `cilk-c`) extends `c` and is named "Cilk/C".
- `cilkcpp` (aliases: `cilk-cpp`, `cilk`) extends `cpp` and is named "Cilk/C++".

### Extra question

If possible, should I specify the `OpenCilk` organization as the "owner"? 

The Prism README states:  "If you contribute a new language definition, you will be responsible for handling bug reports about that language definition."  I assume it would still be primarily on me to deal with this, but giving "ownership" to OpenCilk seems like a better long-term path.

### Additional comments

- The entire implementation of adding Cilk C/C++ support to Prism is only a handful of lines of code, excluding tests/examples.  You can see it [here](https://github.com/ailiop/prism/commit/5446ce32db9fd35bb338e8bf3c674d4663307a83) (see the first 3 files).
- Prism separates syntactic tokenization from thematic highlighting.  The implementation above only addresses tokenization.  Specifically, it applies a custom token label `cilk-keyword` to Cilk keywords.  Themes are then responsible for highlighting tokens based on their label.  Since there is no guarantee that a theme will support non-standard token labels, the recommendation is to specify a standard fallback.  You can find the list of standard token labels [here](https://prismjs.com/tokens.html); I have opted for `important` as the default fallback.
- The Cilk keywords I registered are: `cilk_scope`, `cilk_spawn`, `cilk_sync`, `cilk_for`, and `cilk_reducer`.

### Example with Cilk/C++ (` ```cilkcpp `) and the default theme

![prism-cilk-kwd-example](https://user-images.githubusercontent.com/1577182/183158034-85e8779d-334a-4666-8d67-668b03c8e8c8.png)

**Comment by behoppe** [comment]
Bravo @ailiop.

I am especially curious what @neboat thinks -- I think he built something that worked with Jekyll, but I get the impression that his approach was more complicated? (I could be wrong.) 

Making sure we use the latest PrismJS is the least I can do to help with this.

**Comment by behoppe** [comment]
For what it's worth, here's a figure I just made using the lime green instead of tangelo for Cilk keywords.

![fib-code-multicore-wide](https://user-images.githubusercontent.com/93284810/183985406-c043ec0c-7ef7-4399-b250-8bd7332cf6d4.png)

**Comment by ailiop** [comment]
The `cilkc` and `cilkcpp` languages have been merged into PrismJS!

There are two changes compared to my previous comment which would affect theming:

1. The token label that is applied to the Cilk keywords is `parallel-keyword` (*not* `cilk-keyword`).
2. The fallback "alias" token label for Cilk keywords is `keyword` (*not* `important`).  This means that themes which do not recognize `parallel-keyword` tokens will default to highlighting them like standard keywords.

Thank you @tfk and @neboat for chatting with me about these things last week.

**Comment by ailiop** [comment]
@behoppe I think we should leave this issue open until we also have a theme which highlights `parallel-keyword` tokens to our satisfaction, but I leave this up to you.

**Comment by behoppe** [comment]
Thanks @ailiop. I am excited to deploy this. Unfortunately I am struggling to get the latest Prismjs incorporated into the site. I have reduced my problem down to the following: when I create a new Node server project from scratch and install Prismjs as the only package, I can't get the very latest version with `cilkc` and `cilkcpp`. This same issue happens with the OpenCilk site, resulting in an error message "cilkc is not a valid Prism.js language for eleventy-md-syntax-highlight" from the [Eleventy plugin](https://github.com/pborenstein/eleventy-md-syntax-highlight) we use to interface with Prismjs.

So it seems that I can't convince the Node package manager to use the latest content on the Prismjs GitHub repo. Any ideas?

Below are the steps I followed:
```sh
$ node -v
v12.22.12
$ npm -v
6.14.16
$ mkdir test
$ cd test
$ npm init
This utility will walk you through creating a package.json file.
It only covers the most common items, and tries to guess sensible defaults.

See `npm help init` for definitive documentation on these fields
and exactly what they do.

Use `npm install <pkg>` afterwards to install a package and
save it as a dependency in the package.json file.

Press ^C at any time to quit.
package name: (test) 
version: (1.0.0) 
description: 
entry point: (index.js) 
test command: 
git repository: 
keywords: 
author: 
license: (ISC) 
About to write to /home/bruce/11ty/test/package.json:

{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}


Is this OK? (yes) 

$ npm install prismjs
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN test@1.0.0 No description
npm WARN test@1.0.0 No repository field.

+ prismjs@1.28.0
added 1 package from 1 contributor and audited 1 package in 0.753s
found 0 vulnerabilities

$ npm update prismjs
$ ls node_modules/prismjs/components
```

The `ls` command will reveal that `cilkc` and `cilkcpp` files are not present.

**Comment by ailiop** [comment]
@behoppe I am not too familiar with Node and do not think I can offer much insight.  But I saw you asked the PrismJS folks and if I understand correctly the new PrismJS release will come out this weekend and the problem will then "fix itself", so to speak.  Is this right?  (In other words, I don't need to think about it, correct? 🙂)

Thanks for looking into this!

**Comment by behoppe** [comment]
@ailiop that's right. The PrismJS folks have not released the new version, and there is nothing we can do but wait a few days for that to happen. I guess "release the new version" somehow sends it into the Node package world, where it will make its way into our website.

**Comment by ailiop** [comment]
@behoppe PrismJS just got a new release, [v1.29.0](https://github.com/PrismJS/prism/releases/tag/v1.29.0), a few hours ago.  :)
